python/ray/ml/integrations/torch.py

```diff
@@ -0,0 +1,188 @@
+
+    if multi_input and column_dtypes:
+        if not type(column_dtypes) not in [torch.dtype, list, tuple]:
+            raise TypeError(
+                "If `columns` is a list of lists, "
+                "`column_dtypes` must be None, `torch.dtype`,"
+                f" or a sequence, got {type(column_dtypes)}."
+            )
+
+    if (
+        not multi_input
+        and column_dtypes
+        and type(column_dtypes) not in [torch.dtype, list, tuple]
+    ):
+        raise TypeError(
+            "If `columns` is a list of strings, "
+            "`column_dtypes` must be None or a single `torch.dtype`."
+            f"Got {type(column_dtypes)} instead."
+        )
+
+    def get_tensor_for_columns(columns, dtype):
+        feature_tensors = []
+
+        if columns:
+            batch = data_batch[columns]
+        else:
+            batch = data_batch
+
+        for col in batch.columns:
+            col_vals = batch[col].values
+            t = torch.as_tensor(col_vals, dtype=dtype)
+            t = t.view(-1, 1)
+            feature_tensors.append(t)
+
+        return torch.cat(feature_tensors, dim=1)
+
+    if multi_input:
+        if isinstance(column_dtypes, torch.dtype):
+            column_dtypes = [column_dtypes] * len(columns)
+        return [
+            get_tensor_for_columns(columns=subcolumns, dtype=dtype)
+            for subcolumns, dtype in zip(columns, column_dtypes)
+        ]
+    else:
+        return get_tensor_for_columns(columns=columns, dtype=column_dtypes)
+
+
+class TorchPredictor(Predictor):
+    """A predictor for PyTorch models."""
+
+    def __init__(self, model: torch.nn.Module, preprocessor: Preprocessor):
+        self.model = model
+        self.model.eval()
+        self.preprocessor = preprocessor
+
+    @classmethod
+    def from_checkpoint(
+        cls, checkpoint: Checkpoint, model_definition: Optional[torch.nn.Module] = None
+    ) -> "TorchPredictor":
+        """Instantiate the predictor from a Checkpoint.
+
+        The checkpoint is expected to be a result of ``TorchTrainer``.
+
+        Args:
+            checkpoint (Checkpoint): The checkpoint to load the model and
+                preprocessor from. It is expected to be from the result of a
+                ``TorchTrainer`` run.
+            model_definition (Optional[torch.nn.Module]): If checkpoints
+                contains the model state dict, and not the model itself,
+                then the state dict will be loaded to the model_definition.
+
+        """
+        checkpoint_dict = checkpoint.to_dict()
+        preprocessor = checkpoint_dict[PREPROCESSOR_KEY]
+        model = load_torch_model(
+            saved_model=checkpoint_dict[MODEL_KEY], model_definition=model_definition
+        )
+        return TorchPredictor(model=model, preprocessor=preprocessor)
+
+    def predict(
+        self,
+        data: DataBatchType,
+        column_names: Optional[List[str]] = None,
```

Comment:
These aren't part of the predictor API. Either remove or move to the constructor.

Reply:
Is there a reason not to support **kwargs in `predict`? I feel these arguments are specific to `predict` method itself and not the Predictor object. also cc @simon-mo for thoughts.

Reply:
Let's discuss this with the broader group. I think it may motivate changing the Predictor interface.